VS-RustAnalyzer/Util.cs

```diff
@@ -21,5 +22,26 @@ public static bool IsCargoFile(string path)
         {
             return cargoMatcher.IsMatch(path);
         }
+
+        public static string GetCargoFileForPath(string filePath, string projectRoot)
+        {
+            var currentPath = Path.GetDirectoryName(filePath);
+
+            while (true)
```

Comment:
This could infinitely loop if we pass a path not rooted in projectRoot.
We should have an additional error case either in the loop when we hit the end or some verification that the filePath is a subPath of projectRoot.